.changelog/16097.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
+agent: only use the `agent` token and do not attempt using a service token for internal deregistration of checks and services
```

Comment:
Can you think of a scenario by which someone may view this as a breaking change in behavior that they were relying on?

Reply:
I guess I could phrase this as a breaking change by saying: You now _must_ have an agent token configured with node:write permission for a client agent to deregister services and checks.

(I feel like this was sort of already a sort of implicit requirement for the agent to function correctly though.)

I was also considering `bug-fix` here, since it fixes the case where a service token has been deleted and the agent never deregisters the service.

Reply:
I updated the changelog entry to be a bug-fix now. Let me know what you think!